packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable i18next/no-literal-string */
+import { 
+  Component, 
+  Element, 
+  Host, 
+  h, 
+  Prop, 
+  forceUpdate,
+  Fragment,
+  Event,
+  EventEmitter
+} from '@stencil/core';
+import i18next from 'i18next';
+
+@Component({
+  tag: 'va-file-input',
+  styleUrl: 'va-file-input.css',
+  shadow: true,
+})
+
+export class VaFileInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the file input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button.
+   */
+  @Prop() buttonText: string;
```

Comment:
Why won't this work?

```suggestion
  @Prop() buttonText: string = "Upload file";
```

Reply:
@bkjohnson Sorry if I didn't explain completely. I'm seeing that there is a difference between setting a "default" and "requiring a default". My requirement is to make sure that the button _always_ has a specific default text but also allow it to be dynamic.

If I just did `@Prop() buttonText: string = "Upload file";`, then the prop could be set to blank and the button would have no text like this:

![Screen Shot 2022-08-17 at 1 03 26 PM](https://user-images.githubusercontent.com/872479/185210791-5aefcf67-a4c9-4822-a661-924cff89a46f.png)

The utility function allows the value to be dynamic while also enforcing a default value.

Reply:
Gotcha :+1: